grails-app/services/com/asaas/mini/CustomerService.groovy

```diff
@@ -121,20 +123,49 @@ class CustomerService {
     }
 
     public void delete(Long id) {
-        Customer customer = Customer.findByIdAndDeleted(id, false)
+        if (!id) {
+            throw new IllegalArgumentException("ID is required")
+        }
 
+        Customer customer = Customer.findByIdAndDeleted(id, false)
         if (!customer) {
             throw new IllegalArgumentException("Conta não encontrada")
         }
+
         validateDelete(customer)
+
+        List<Payer> payers = Payer.findAllByCustomerAndDeleted(customer, false)
+        payers.each { payer ->
```

Comment:
no asaas a gente n usa each, pois ele n agrega na em relacao ao for e cria o problema de cada item ser processado dentro de uma closure, entao pode gerar confusao se vc fizer algo como return dentro desta closure (no each vai seguir para o proximo item, no for vai parar e sair do metodo atual).